### ISSUE TITLE
Prepare release 3.7.0

### ISSUE THREAD

**Issue by hoffie** [jamulussoftware/jamulus#1086]
- [x] Decide if this should be a bugfix release (e.g. 3.6.2 -> 3.6.3) or a feature release (3.6.2 -> 3.7.0). -> We have a feature release, so it's 3.7.0
- [x] Create a milestone and add all relevant issues (@ann0see did this)
- [x] Ensure that all issues/PR targeted for this release are done by checking the [milestone](https://github.com/jamulussoftware/jamulus/milestone/1)
- [x] Declare a code freeze. PRs can still be worked on and may get reviewed, but must not be merged unless agreed explicitly.
- [x] Create at least one beta release (@ann0see did that). **WARNING**: Do not deploy beta releases to the central server which is used for update checks.
- [x] Get feedback on the stability of the beta release (@pljones and @hoffie are running servers and/or clients on the beta level)
- [x] Website translations
  - [x] Create new branch `translate3_7_0` based on branch `release` (@ann0see did this)
  - [x] Squash-merge `changes` into `translate3_7_0`
  - [x] Ping translators. Translators should open PRs with their changes against the `translate3_7_0` branch (?) and reference this issue ("Add German translations for 3.7.0")
  - [x] Wait for all PRs to be merged
  - [x] Merge `translate3_7_0` into release
- [x] App installer translations
  - [x] Ping translators. @ann0see has done this individually?
  - [x] Review PRs, <details><summary>Checklist for the PRs</summary>
    ```
    - [ ] Translator listed in the `src/util.cpp` **optionally add link to PR or code**
    - [ ] Looks consistent
    - [ ] Passes `tools/check-wininstaller-translations.sh`
    ```
    </details>
  - [x] Wait for all PRs to be merged:
    - pl #1124
    - fr #1125
    - es #1152
- [x] App translations (@softins handles this currently, https://github.com/orgs/jamulussoftware/teams/translators/discussions/6/comments/4)
  - [x] Generate `.ts` files in master via `lupdate` (@softins, directly in master)
  - [x] Ping translators (@softins). Translators do their work using `linguist` and submit PRs with their changes against master. Get the list of translators from `src/utils.cpp` (some grep/sed would be nice; last one per language is the active one).
  - [x] Review PRs, <details><summary>Checklist for the PRs</summary>
     ```
     - [ ] Translator listed in the `src/util.cpp` **optionally add link to PR or code**
     - [ ] Punctuation and spacing consistent
     - [ ] Signal words consistent ("ASIO", "Buffer")
     - [ ] No untranslated strings (`grep unfinished -5 src/res/translation/translation_$TRANSLTION*.ts`)
     - [ ] Only a single `.ts` file checked in
     ```
     </details>
  - [x] Wait for all PRs to be merged:
    - de_DE #1097
    - es_ES #1152
    - fr_FR #1113
    - it_IT #1112
    - nl_NL #1110
    - pl_PL #1099
    - pt_BR #1141
    - pt_PT #1168 #1208
    - sk_SK #1104
    - sv_SE #1106
  - [x] Generate `.qm` files via `lrelease Jamulus.pro`
- [x] Tag a release candidate
- [x] Draft an announcement, include all contributors via `tools/get_release_contributors.py` (#1223)
- [x] Update the version number in `Jamulus.pro` and add the release date to the Changelog header and commit
- [x] Tag this commit as r3_7_0
- [x] Prepare `Jamulus.pro` (`dev` suffix) and ChangeLog (add a header) for the next release
- [x] Wait for the build to complete
- [x] Do a smoke test for Windows/Mac/Linux -- Do the binaries start/connect properly?
- [x] Upload the artifacts to SourceForge and link latest
- [x] Trigger the update notification by updating the Old Default Central Server
- [x] Announce the new release with a summary of changes (+ link to the changelog for details), a link to the download page
  - [ ] ~On Sourceforge? (only if it has been done in the past?)~
  - [x] On Github discussions? (An "Announcements" sub-form maybe?)
  - [x] On Facebook in the group "Jamulus (official group)"
- [x] Update download links on the website (start page, per-platform pages + translations) https://github.com/jamulussoftware/jamuluswebsite/issues/359
- [x] Update this checklist with any improvements which were found
----
As this is the first time we use such an checklist
- [ ] Decide where to keep the release checklist:
  - Option 1: Make it an issue template. Would make it simple for us, but could be confusing for non-developers.
  - Option 2: Add it to the Wiki
- [ ] Should one part of the checklist live in the website repo?

Please feel free to edit/extend this issue description! Just trying to document what needs to be done as we move along.

**Comment by ann0see** [comment]
> Update the version number in Jamulus.pro and commit

After this commit, the version entry to the changelog has to be created.

**Comment by pljones** [comment]
There's something missing.

In order to trigger the update notification in the client, the (old) "Default Central Server" needs updating -- that should happen just _after_ publishing the artifacts (from the published artifact for the platform, ideally).

**Comment by gilgongo** [comment]
I think the website translation steps make sense to me now. I guess at the end of the process we delete the translation branch?

**Comment by atsampson** [comment]
It might be a good idea to explicitly add "Test the generated artifacts" before "Upload the artifacts to Sourceforge" - i.e. check that the source tarballs are buildable on a real system, and that the binaries can at least start up and connect to a server.

The CI tests will catch some types of errors, but they won't detect problems in the artifact generation process itself, and I've seen plenty of examples over the last few years of projects shipping broken release artifacts when their CI passed (e.g. because the generation process was missing some files, or there was something broken about the tools or libraries in the automated build environment).

**Comment by pljones** [comment]
> at least start up and connect to a server.

... and that we won't break the directory servers when the build gets deployed to them.

**Comment by ann0see** [comment]
>  I guess at the end of the process we delete the translation branch?

@gilgongo 

Yes. That’s what I think should happen.

**Comment by gilgongo** [comment]
OK, so are we good for squash merging changes into translate3_7_0? I think the only new feature for the docs was the `--serverpublicip` description which is done as far as I know. Do we need to write anything up for the `--ctrlmidich` syntax?

**Comment by hoffie** [comment]
@pljones @atsampson @ann0see @gilgongo Thanks for your comments, I have integrated all those suggestions.

> OK, so are we good for squash merging changes into translate3_7_0?

I *think* yes, but I'm certainly not the most qualified to judge that.

> Do we need to write anything up for the --ctrlmidich syntax?

If it changes existing syntax/compatibility, I'd say we must, if it's "only" an extension I'd say it's nice-to-have. Maybe @dakhubgit can help?

**Comment by pljones** [comment]
> Do we need to write anything up for the --ctrlmidich syntax?

Existing syntax should continue to work but without the new features.  The command line help documents the new syntax, I think the web site table got updated, too?

**Comment by gilgongo** [comment]
Yes it did by the looks of it. So I'd say we can merge for translation then. I'll do that this evening and message the translators.

**Comment by softins** [comment]
I have updated the `.ts` translation files for 3.7.0 and committed directly to master. As they are not code, and are automatically generated by `lupdate`, I didn't consider they needed review.

**Comment by gilgongo** [comment]
Just checking since this is the first time I've done this with all the tests running ... The "_Only 65535 characters are allowed_"  remark-lint errors: can I ignore them?

**Comment by ann0see** [comment]
Yes, you can. I'd like to merge my PR before the translations of the website start.

**Comment by gilgongo** [comment]
Oops I didn't see the second part of your message! Can you merge that to translation?

**Comment by ann0see** [comment]
Yes. Just wait a moment.

**Comment by pljones** [comment]
Quick question: we've taken one memory leak bug fix a Mac-specific change to master since beta3 - do how many changes warrant a new beta?  What I'm thinking is I'd like to be testing with beta-tagged code that *could* become latest-tagged with no code commits, and at the moment there are outstanding commits (one that I'd very much like to have running...).

```
F   UID     PID    PPID PRI  NI    VSZ   RSS WCHAN  STAT TTY        TIME COMMAND
4   999    1264       1 -21   - 462444 55196 -      Ssl  ?         67:55 /opt/Jamulus/bin/Jamulus-server -s -n -F -T -p 54850 -u 50 -e jamulusallgenres.fischvolk.de:22224 -m /opt/Jamulus/run/status.html -l /opt/Jamulus/log/Jamulus.log -L -R /opt/Jamulus/run/recording -o jamulus.drealm.info;London;224 -w /opt/Jamulus/systemd/welcome-server.html
```
```
F   UID     PID    PPID PRI  NI    VSZ   RSS WCHAN  STAT TTY        TIME COMMAND
4   999  648187       1 -21   - 240960 55152 -      Ssl  ?          0:00 /opt/Jamulus/bin/Jamulus-server -s -n -F -T -p 54850 -u 50 -e jamulusallgenres.fischvolk.de:22224 -m /opt/Jamulus/run/status.html -l /opt/Jamulus/log/Jamulus.log -L -R /opt/Jamulus/run/recording -o jamulus.drealm.info;London;224 -w /opt/Jamulus/systemd/welcome-server.html
```

**Comment by ann0see** [comment]
@gilgongo done that and opened a new pr

**Comment by softins** [comment]
Hi translators! The .ts files have now been updated from the code for v3.7.0.
Please could you each have a look at your language and provide the updated translations?
There are several new texts to be translated.

Thanks!

@ignotus666 @trebmuh @Snayler @melcon @jerogee @dzpex @rolamos @Martyna27 @SeeLook @genesisproject2020 @jose1711

Note - we have already received PRs for German and Polish - thank you!

**Comment by pljones** [comment]
It's worth resolving
https://github.com/jamulussoftware/jamulus/pull/1097#discussion_r581265793
quickly for the translators.

**Comment by softins** [comment]
> It's worth resolving [#1097 (comment)](https://github.com/jamulussoftware/jamulus/pull/1097#discussion_r581265793) quickly for the translators.

I would agree to leave "The Jamulus Development Team" untranslated, as it is effectively the name of the team.

**Comment by jerogee** [comment]
In English, for QString ```strSndCrdBufDelay``` and ```strSndCrdBufDelayTT```:

> If the buffer delay settings are disabled, it is prohibited by the audio driver to modify this setting from within the software. On Windows, press the ~~ASIO Device Settings~~ button to open the driver settings panel. On Linux, use the Jack configuration tool to change the buffer size.

I do not have Windows to test this, but it should be the "ASIO Setup" button according to [our manual](https://jamulus.io/wiki/Software-Manual).

**Comment by pljones** [comment]
Anyone not agree?  Can anyone see benefit from localisation?

**Comment by gilgongo** [comment]
It's a tricky one. So if Jamulus was an Albanian project, we'd be saying "Brought to you by Ekipi i Zhvillimit të Jamulus"? That's probably OK, but would "Please contact Ekipi i Zhvillimit të Jamulus if you want to get involved" be weird? Usually things like that get translated in English. Maybe it's different for other languages?

**Comment by ann0see** [comment]
Hmm. Good point. What about just letting the translators decide themselves? I think they are the people who know what to do in their language.

**Comment by hoffie** [comment]
> but would "Please contact Ekipi i Zhvillimit të Jamulus if you want to get involved" be weird?

Good point, indeed. "Kontaktiere The Jamulus Development Team" would be weird in German. "Kontaktiere das Jamulus Development Team" would be OK again, although "Kontaktiere das Jamulus-Entwickler-Team" would sound more natural. So when used in a sentence, I'd argue it should be translated. So if we strive for consistency, then it should be translated in the copyright as well.

Just checked what some other Open Source projects are doing:
- Chromium (translated): `Copyright 2021 Die Chromium-Autoren. Alle Rechte vorbehalten.`
- nautilus (untranslated): `© 1999–2018 The Files Authors`
- K-9 Mail (both): `Autoren: Google, das K9-Team und viele Weitere` vs. `© The K-9 Dog Walkers`
- Mozilla website (translated): ` Teile dieses Inhalts sind ©1998–2021 von einzelnen Mitwirkenden von mozilla.org. Inhalt verfügbar unter einer Creative Commons-Lizenz . `

> What about just letting the translators decide themselves? I think they are the people who know what to do in their language.

This would work for me as well.

**Comment by pljones** [comment]
The only one I'm even vaguely concerned about is the copyright string.  Any other usage is definitely not a problem translated.  I'm only worried from a perspective of ensuring the licence remains valid - either where the string _should_ be translated or _should not_ be translated... and I don't know which!

**Comment by melcon** [comment]
> I would agree to leave "The Jamulus Development Team" untranslated, as it is effectively the name of the team.

As @softins said, we could treat it as the name of the team, so no need to translate names.

**Comment by dakhubgit** [comment]
Melcon Moraes <notifications@github.com> writes:

>> I would agree to leave "The Jamulus Development Team" untranslated,
>> as it is effectively the name of the team.
>
> As @softins said, we could treat it as the name of the team, so no
> need to translate names.

We are sort of in the "everything has been said, just not yet by
everybody" stage of the discussion.  There is no formal entity "The
Jamulus Development Team" defined anywhere in the Jamulus source tree
(the only occurence of "development team" in any casing is in the UI
copyright string), so there is no point in treating it as a formal term.
You'll not find any definition of the term anywhere, so sticking with
what is a foreign language to the translation readers serves no purpose.

The only legally sensible meaning without further definition is "all
rights remain with the respective authors" since the whole purpose of a
license is to tell who is going to potentially sue in case of a license
breach.  Introducing a formal term suggests that there is a single
entity taking responsibility and getting the copyright assigned to for
that purpose.  As far as I understand, that is not the case.

So in the case of German, I'd use "das Jamulus-Entwickler-Team" (note
that I don't capitalise "das" in order not to suggest a formal term)
which would be the German spelling typically employed here, possibly
"das Jamulus-Entwicklerteam" which is a bit more classic German style
(using hyphens only to separate significantly separate terms, not word
compounds).

But that's the choice of the active translator: no point in introducing
style inconsistencies.

-- 
David Kastrup

**Comment by hoffie** [comment]
@Snayler Friendly re-ping, are you planning to update the pt_PT translation for 3.7.0? I'm asking because (I think) we are close to a last beta release.

**Comment by pljones** [comment]
> Tag this commit as r3_7_0
> Prepare Jamulus.pro (dev suffix) and ChangeLog (add a header) for the next release

Has it been tested now that tagging will generate the required artifacts with the correct version string?  i.e. that no manual change to Jamulus.pro to get the version is required for a release build?

**Comment by hoffie** [comment]
> > Tag this commit as r3_7_0
> > Prepare Jamulus.pro (dev suffix) and ChangeLog (add a header) for the next release
> 
> Has it been tested now that tagging will generate the required artifacts with the correct version string? i.e. that no manual change to Jamulus.pro to get the version is required for a release build?

I don't think there have been any changes in this area. There is no automation yet. The previous checklist item contains the instruction to update `Jamulus.pro` manually and I think that's fine for 3.7.0.

So, to clarify:
- The main developer preparing the final release will have to do this manually (as it used to be).
- People who want to build the release from the git tag, do not have to do anything manually.

To be extra sure, I'm running a release test in my own repo. I have [updated](https://github.com/hoffie/jamulus/commit/9952fccaba0a40a7591ebdf4a9c34643e3d9e942) `Jamulus.pro` to 0.0.0.3.7.0 there (to avoid any confusion with official releases) and tagged it as [r0_0_0_3_7_0](https://github.com/hoffie/jamulus/tree/r0_0_0_3_7_0). CI is running and we can check the reported version of the binaries later at https://github.com/hoffie/jamulus/actions/runs/617020857

**Comment by softins** [comment]
I don't see anything in the checklist about tagging the release also with "latest". This needs to be done, as there are build scripts out there that rely on it. Volker added it as part of the release process a few versions ago. We don't need artifacts built with "latest" in the filename, just to have the tag available for people to checkout. I'm not 100% sure how that works in practice. I think it's that r3_7_0 is tagged as an actual release for 3.7.0, but latest is just a tag pointing at the same point in the repo.

**Comment by pljones** [comment]
Yep, looks okay now 👍 .

**Comment by pljones** [comment]
> We don't need artifacts built with "latest" in the filename, just to have the tag available for people to checkout. I'm not 100% sure how that works in practice. 

`git checkout r3_7_0; git tag --force latest; git push --force --tags` on jamulussoftware/jamulus/master, I think.

**Comment by pljones** [comment]
>  - [ ] Review PRs, Checklist for the PRs
...
>     - [ ] No untranslated strings (`grep unfinished -5 src/res/translation/translation_$TRANSLTION*.ts`)

I can see a whole bunch of `<translation type="vanished">Compartilhar</translation>` entries.  Anyone have a clue what they are?  Should they exist?  As another more concrete example,
```
src/res/translation/translation_de_DE.ts:        <source>Country flag icons from Mark James</source>
src/res/translation/translation_de_DE.ts-        <translation type="vanished">Die Bilder der Länderflaggen sind von Mark James</translation>
```
The `source` doesn't get found in `src/*.{h,cpp}` -- so perhaps these are "vanished" and need to be deleted?

**Comment by hoffie** [comment]
> so perhaps these are "vanished"

Yes, I think those were used in previous versions.

> and need to be deleted?

I think the idea is to keep them as a help if an identical (or even similar?) string will need to be translated in a future version. They do not end up in any release artifacts, as far is I understand.

https://stackoverflow.com/questions/31808275/vanished-type-for-the-translation-element-in-the-ts-files/31892150#31892150

**Comment by Snayler** [comment]
> 
> 
> @Snayler Friendly re-ping, are you planning to update the pt_PT translation for 3.7.0? I'm asking because (I think) we are close to a last beta release.

Sorry, I've been away for some time now. I'm working on it and will have it ready ASAP.

**Comment by softins** [comment]
> > so perhaps these are "vanished"
> 
> Yes, I think those were used in previous versions.
> 
> > and need to be deleted?
> 
> I think the idea is to keep them as a help if an identical (or even similar?) string will need to be translated in a future version. They do not end up in any release artifacts, as far is I understand.
> 
> https://stackoverflow.com/questions/31808275/vanished-type-for-the-translation-element-in-the-ts-files/31892150#31892150

Yes, that's exactly right. The vanished and unfinished entries do not get put into the `.qm` files.

I did consider doing `lupdate -no-obsolete Jamulus.pro`, which would discard all the obsolete or vanished entries when updating the `.ts` files, but I decided against it, as they could be a useful reference in the future.

**Comment by ann0see** [comment]
If I see correctly all translations are finished and we‘re ready for the release candidate?

**Comment by softins** [comment]
> If I see correctly all translations are finished and we‘re ready for the release candidate?

Still waiting for sk_SK to be finished, I believe.

**Comment by hoffie** [comment]
> If I see correctly all translations are finished and we‘re ready for the release candidate?

I'd say yes. What would this version be called? We only talked about having additional "beta" releases so far. You mentioned the phrase "release candidate" and I think this accurately describes it. Other projects tag such releases as "rc". Should we als tag it as 3.7.0rc1 (with the Jamulus.pro version still being kept as dev-XXX until @pljones's optimization lands)?
@pljones: Note for the future (not relevant for 3.7.0): The "r" in "rc" may interfere with the tag-to-version conversion based on string replacements.

> Still waiting for sk_SK to be finished, I believe.

I understood this to be "on my TODO, but not sure when it gets ready". So unless Jose comments otherwise, I'd suggest to move forward. As far as I remember the untranslated strings are nothing new. So even if we released in a few days, it would not make things worse for sk_SK users.

**Comment by ann0see** [comment]
There's still #1148 and after this release an investigation of the ASIO API in Jamulus https://github.com/jamulussoftware/jamulus/pull/990

**Comment by softins** [comment]
#1148 was merged yesterday. I agree that unless sk_SK is done very soon, we just go to a 3.7.0rc1. I have already done a `lrelease` and committed the generated `.qm` files from the latest translations.

I think the idea of release candidates is that the last rc (which may even be rc1) gets tagged as the actual release with no further code changes (except for the version in `Jamulus.pro`).

**Comment by pljones** [comment]
> Update the version number in Jamulus.pro and add the release date to the Changelog header and commit

After that but before

> Tag this commit as r3_7_0

should we have a smoke test to make sure all assets being released are correct and function as expected?  Or is the tag needed (I'm now utterly confused about the autobuild process...)?  Is "latest" needed, too?

**Comment by hoffie** [comment]
> should we have a smoke test to make sure all assets being released are correct and function as expected?

Definitely. This was suggested before and I'd added it, but somehow it got lost. I just restored it (and it changed formatting of the list and I can't say why; maybe Github adds spacing when the list gets really long?).

> Or is the tag needed (I'm now utterly confused about the autobuild process...)?

Adding a tag and pushing it starts the final release build, so this is required before testing. We could/should perform the smoke test on the release candidate instead which must not be different code-wise.

@softins You modified that part recently (split into tag + build release). Is this really necessary? If so, what's the difference?

> Is "latest" needed, too?

I think it's only needed for people directly checking out "latest". So yes, we should probably do it, but it's not relevant for our tests.

**Comment by ann0see** [comment]
Ok. So should we do the rc 1 release now? (Note: the tag needs to include the word beta to flag the release as pre release on GH).

**Comment by hoffie** [comment]
> Ok. So should we do the rc 1 release now?

Code-wise I'd say yes. Let's work out the naming first though.

> (Note: the tag needs to include the word beta to flag the release as pre release on GH).

So this would need to be 3.7.0beta4_rc1 or something? This would sound really strange to me.
I guess it is related to [this logic](https://github.com/jamulussoftware/jamulus/blob/master/.github/actions_scripts/analyse_git_reference.py#L78) in the autobuild action_scripts/analyse_git_reference.py? I think this logic should be adapted in the same way to what @softins did for the update check: Consider all versions without suffix a final release, consider everything else non-final.

I think we should either make it possible to create a 3.7.0rc1 now or we should go with 3.7.0beta4 due to the technical constraints, mark it as release candidate in an informal way (I guess we can edit the release notes which are auto-generated from the Changelog?) and adapt the build logic later. I'd prefer the former, but would be OK with the latter as well.

**Comment by softins** [comment]
> @softins You modified that part recently (split into tag + build release). Is this really necessary? If so, what's the difference?

I found that when I created the tag on my local system and then pushed it to origin, when I looked at github, the tag was there with only 2 assets: the sources zip and the sources tar.gz. I then needed to make a release from the tag to get it to build the other assets.

Maybe it is possible to create the tag directly on github when creating the release, all as one operation? I haven't tried that yet.

**Comment by hoffie** [comment]
> I found that when I created the tag on my local system and then pushed it to origin, when I looked at github, the tag was there with only 2 assets: the sources zip and the sources tar.gz. I then needed to make a release from the tag to get it to build the other assets.

Hm, my experience is different, see here: https://github.com/hoffie/jamulus/releases
I haven't performed any operations on Github there, this was simply from tag pushes and I think this is how it should be.

Note: The releases show up pretty quickly, even with the two initial assets. It takes more time until the build has finished (at least ~20mins?). After that, the assets will show up one by one. Maybe that was the reason?

**Comment by softins** [comment]
> > I found that when I created the tag on my local system and then pushed it to origin, when I looked at github, the tag was there with only 2 assets: the sources zip and the sources tar.gz. I then needed to make a release from the tag to get it to build the other assets.
> 
> Hm, my experience is different, see here: https://github.com/hoffie/jamulus/releases
> I haven't performed any operations on Github there, this was simply from tag pushes and I think this is how it should be.
> 
> Note: The releases show up pretty quickly, even with the two initial assets. It takes more time until the build has finished (at least ~20mins?). After that, the assets will show up one by one. Maybe that was the reason?

Yes, that may well be. It was the first time I had tried it, so I wasn’t certain what to expect.

**Comment by trebmuh** [comment]
French translation update on #1199 
Please merge it before 3.7.0 goes public.

**Comment by hoffie** [comment]
- When #1210 is reviewed and merged, I don't see any blockers for rc1 anymore.
- Not sure if @softins still wanted to check some newly added/updated translation for #1161?
- Sadly, I don't think we will be able to find out more about #1129 in time. Maybe we'll hear back before the final release.

I will only be active tomorrow evening again, feel free to get rc1 out. :)

**Comment by softins** [comment]
> * Not sure if @softins still wanted to check some newly added/updated translation for #1161?

I checked and found one accelerator needed fixing for Swedish. For such a simple change, I committed directly to master, and updated the qm files for FR and SE at the same time.

Just waiting on a review for #1210 to be merged, then I think we are good to go.

* I was thinking, so long as it doesn't get put on Volker's default server, there is no reason we couldn't put 3.7.0rc1 into `Jamulus.pro` for the rc1 release. Do you guys agree?

**Comment by ann0see** [comment]
I do not think that changing the version to 3.7.0rc1 in Jamulus.pro and not deploying it anywhere causes problems. I can deploy it to a custom central server I set up (which is also checked for updates) and test it.

**Comment by ann0see** [comment]
Ok. Something is wrong with the auto build stuff. It creates a draft release. It's ok that it doesn't find a change log but not the draft thing.

**Comment by ann0see** [comment]
I think https://github.com/jamulussoftware/jamulus/pull/1194#issuecomment-794502021

Is the last thing before the rc tag. I'd say should try to get the rc out today/tomorrow and the release in the next few days. The process already lasted almost about 3 weeks...

**Comment by softins** [comment]
3.7.0rc1 now tagged and pushed. Waiting for the assets to build.

**Comment by ann0see** [comment]
Great! I just re ran the check since it seemed to fail on Windows. Not sure why.

**Comment by trebmuh** [comment]
(_Preaching for my own church here_)

Could the translations updates be a part of the [release notes](https://github.com/jamulussoftware/jamulus/discussions/1217#discussion-3261926)?
It would be appreciated since it's a real contribution to Jamulus, even if it's not coding.

Thanks.

**Comment by dakhubgit** [comment]
Olivier Humbert ***@***.***> writes:

> (_Preaching for my own church here_)
>
> Could the translations updates be a part of the [release
> notes](https://github.com/jamulussoftware/jamulus/discussions/1217#discussion-3261926)?
> It would be appreciated since it's a real contribution to Jamulus,
> even if it's not coding.
>
> Thanks.

And it's grunt work entirely dependent on the decisions of others.  The
kind of stuff I myself find rather hard to bring myself to.  And like
the original documentation writing (and the best documentation also
tends to be written after the fact, from people not invested into
defending the programmers' decisions as much as making the user able to
work with them) it makes a world of difference regarding bringing an
end-user oriented application to, well, the end-user.

So proper credit here is certainly a good thing.

-- 
David Kastrup

**Comment by ann0see** [comment]
> Could the translations updates be a part of the release notes (and here too)?

Sure. At least for me this would be ok. Not sure what others think.

**Comment by hoffie** [comment]
> Could the translations updates be a part of the [release notes](https://github.com/jamulussoftware/jamulus/releases/tag/r3_7_0rc1) ([and here too](https://github.com/jamulussoftware/jamulus/discussions/1217#discussion-3261926))?
> It would be appreciated since it's a real contribution to Jamulus, even if it's not coding.

Yes, the translations are certainly valuable and very important for the success of Jamulus! I agree that the Changelog does not do a good job for crediting the people doing the continuous grunt work.

I'd suggest the following:
- The 3.7.0 Changelog lacks attributions for the **new** translations (so, mainly the installer stuff). This should be fixed in the Changelog/release notes. I will add this as part of #1218.
- I hope that we will have a more generic, less technical release announcement. I plan to write something up for that. I think this could be the place for general mentions of translators, contributors and main developers.
- We could have some "The people behind Jamulus" page which describes our project structure and maybe links to the Github org's People page (to avoid having to constantly update the page). If someone wants to make this happen, I suggest opening a discussion/issue for the website.

@trebmuh Would this work for you?

**Comment by trebmuh** [comment]
Sounds good.

What about adding something like this to the release notes too:
* GUI translation update:
  * French (by @trebmuh & @jujudusud) #1199 & #1113
  * Swedish (by @genesisproject2020) #1182 & #1106 
  * Portugese (by @melcon & @Snayler) #1141 & #1168
  * Spanish (by @ignotus666) #1152
  * Italian (by @dzpex) #1112
  * Dutch (by @jerogee) #1110
  * Slovak (by @jose1711) #1104
  * Polish (by @SeeLook) #1099
  * German (by @rolamos) #1097

**Comment by ann0see** [comment]
I think the only blocking thing for the next release is the bug I experienced. Since nobody else seems to be able to reproduce it, maybe it's invalid?

**Comment by softins** [comment]
> I think the only blocking thing for the next release is the bug I experienced. Since nobody else seems to be able to reproduce it, maybe it's invalid?

So much has been happening, I've forgotten what that is?

**Comment by ann0see** [comment]
It's in the release announcement post. Basically there seem to be graphical glitches on Debian 10 Gnome while switching between different skins

**Comment by hoffie** [comment]
Can anyone confirm that a full `lrelease Jamulus.pro` has been run before 3.7.0rc1 was tagged? I see several commits for specific `.qm` files, but no update-them-all. When I'm running that command, all files get changed (might include a timestamp or I may be running a different version), so I can't verify it by doing that.

If we are unsure, should we do it before the final release (without another RC)?

**Comment by softins** [comment]
> Can anyone confirm that a full `lrelease Jamulus.pro` has been run before 3.7.0rc1 was tagged? I see several commits for specific `.qm` files, but no update-them-all. When I'm running that command, all files get changed (might include a timestamp or I may be running a different version), so I can't verify it by doing that.
> 
> If we are unsure, should we do it before the final release (without another RC)?

Yes, I can confirm. `lrelease` may well update all qm files, but git will notice which one's contents have actually changed. At several points leading up to 3.7.0rc1 I did a full `lrelease Jamulus.pro` and then committed those specific .qm files that had changed. I think just before setting the tag, I checked whether any updates to the .ts files had been done since I last checked in a .qm, and found there hadn't been.

There is no harm in doing a new `lrelease Jamulus.pro` before final release. I can do that. There should be no change in the qm files (provided it was done on the same system - there might be minor variations between architecture or release of Qt).

**Comment by hoffie** [comment]
> but git will notice which one's contents have actually changed

Yes, that's why I was confused. git shows all 10 .qm files as modifed. I expected to show up none (or only a few files) to show up.

> There is no harm in doing a new lrelease Jamulus.pro before final release. I can do that. There should be no change in the qm files

Yes, that would be good.

>(provided it was done on the same system - there might be minor variations between architecture or release of Qt)

Yep, this might be the reason for the above observation.

**Comment by ann0see** [comment]
Ok. I think finally we're ready. Both PRs seem to be ok?

**Comment by ann0see** [comment]
Just merged my PR.

**Comment by pljones** [comment]
https://github.com/jamulussoftware/jamuluswebsite/pull/333 is "Waiting Externally" - what's it waiting for?

**Comment by hoffie** [comment]
> [jamulussoftware/jamuluswebsite#333](https://github.com/jamulussoftware/jamuluswebsite/pull/333) is "Waiting Externally" - what's it waiting for?
> 
> Does it some how relate to [jamulussoftware/jamuluswebsite#341](https://github.com/jamulussoftware/jamuluswebsite/pull/341) ?

Yes, jamulussoftware/jamuluswebsite#333 is waiting for jamulussoftware/jamuluswebsite#341. As jamulussoftware/jamuluswebsite#341 is waiting for translators to confirm (?), I chose "Waiting Externally". Happy to hear about better ways to visualize that. ;)

**Comment by pljones** [comment]
It's a "meta-issue" - i.e. it's controlling a number of sub-tasks, always tricky to handle.  Perhaps left in "In progress" with a comment tagging the requested reviewers stating the ticket would not progress without their sign off?  (It's basically "code" with a pull request open waiting requested reviewers...)

**Comment by ann0see** [comment]
Yes. I think rc2 would be good. Release on Wednesday or even earlier?

**Comment by softins** [comment]
Shall I do the rc2 release this evening, using the same procedure as I did for rc1?

**Comment by ann0see** [comment]
Yes. I think that would be great

**Comment by softins** [comment]
OK, I've pushed the tag r3_7_0rc2. Just waiting for it to be made into a release by github-actions.
I've also further updated Jamulus.pro to 3.7.0rc2dev.

**Comment by ann0see** [comment]
Ok. So what is still missing for Release 3.7.0?
macOS, Linux and Windows seem to work. I haven't heard about any serious bugs either. 

My suggestion would be to release either today evening or tomorrow. 

Any other options?

I think my issue with the dev files can wait.

**Comment by hoffie** [comment]
> Ok. So what is still missing for Release 3.7.0?
> macOS, Linux and Windows seem to work. I haven't heard about any serious bugs either.
> 
> My suggestion would be to release either today evening or tomorrow.

Yes, I also suggested today evening in the chat. @softins agreed as well.

**Comment by jerogee** [comment]
Great work! Looking forward to 3.7.0 getting out.

**Comment by gilgongo** [comment]
Just noting what I think we agreed in the chat: 

So as to enable better visibility on the status of translations, it may be best to open issues for each language on both repos (probably also assigned to the relevant people?) so that we can track these. When each PR comes in we/they can attach them to the issue so they both get closed out on merge. Bonus points for a GH action that creates the relevant issues in bulk :-)

This would also mean the "ping" would be done by the assignment of the issue and any questions before the PR comes in can be dealt with easier.

**Comment by pljones** [comment]
Also from chat:

For SourceForge:
- you need an authorised SourceForge account
- ftp accname,llcon@frs.sourceforge.net
- cd /home/pfs/project/l/ll/llcon

You'll see "Jamulus" and "OldFiles" directories.  "Jamulus" is the releases, one directory per release.
- cd Jamulus
- mkdir 3.7.0
- cd 3.7.0
- Upload all the files.

Out of FTP and on SourceForge "Files", for each of the files uploaded, click the _[i]_ icon next to the file and set it as "Default for (appropriate) platform)"

(Note that unsetting the "default" - so none is set - automatically picks the newest, so don't do that -- explicitly set the new default when the release announcement is posted.  This lets all the files be uploaded in advance ready.)

**Comment by pljones** [comment]
Also, I'd suggest we close this discussion and start a "post-3.7.0 Improvements to release process", so there's no open 3.7.0 tickets :)

**Comment by hoffie** [comment]
Volker has updated the default central server, so I'm closing this. I will also close the 3.7.0 milestone shortly so it doesn't show up when triaging new issues.

I'll take care of documentation here: https://github.com/jamulussoftware/jamuluswebsite/issues/340
I'll open a discussion about a quick retrospective later.
